featurebyte/worker/task/observation_table.py

```diff
@@ -188,15 +199,94 @@ async def create_observation_table(
+                payload=payload,
+            ):
+                # compute observation table without target column first
+                await request_input.materialize(
+                    session=db_session,
+                    destination=temp_location.table_details,
+                    sample_rows=payload.sample_rows,
+                    sample_from_timestamp=payload.sample_from_timestamp,
+                    sample_to_timestamp=sample_to_timestamp,
+                    columns_to_exclude_missing_values=columns_to_exclude_missing_values,
+                    missing_data_table_details=missing_data_table_details,
+                )
+                try:
+                    additional_metadata = await self.observation_table_service.validate_materialized_table_and_get_metadata(
+                        db_session,
+                        temp_location.table_details,
+                        feature_store=feature_store,
+                        skip_entity_validation_checks=payload.skip_entity_validation_checks,
+                        primary_entity_ids=payload.primary_entity_ids,
+                        target_namespace_id=target_namespace_id,
+                    )
+
+                    # compute observation table with target column
+                    observation_set = ObservationTableModel(
+                        user_id=payload.user_id,
+                        name=payload.name,
+                        location=temp_location,
+                        context_id=context_id,
+                        use_case_ids=use_case_ids,
+                        request_input=payload.request_input,
+                        purpose=payload.purpose,
+                        primary_entity_ids=payload.primary_entity_ids or [],
+                        **additional_metadata,
+                    )
+
+                    result = await self.target_computer.compute(
+                        observation_set=observation_set,
+                        compute_request=ComputeTargetRequest(
+                            feature_store_id=payload.feature_store_id,
+                            graph=graph,
+                            node_names=node_names,
+                            target_id=to_compute_target_id,
+                        ),
+                        output_table_details=location.table_details,
+                    )
+                    is_view = result.is_output_view
+                finally:
+                    # drop temporary table
+                    await db_session.drop_table(
+                        table_name=temp_location.table_details.table_name,
+                        schema_name=temp_location.table_details.schema_name
+                        or db_session.schema_name,
+                        database_name=temp_location.table_details.database_name
+                        or db_session.database_name,
+                        if_exists=True,
+                    )
```

Comment:
Possible to refactor this into a smaller method since `create_observation_table` is getting very long?